packages/runtime/src/useCases/consumption/attributes/DeleteRepositoryAttribute.ts

```diff
@@ -0,0 +1,68 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController, LocalAttribute } from "@nmshd/consumption";
+import { AccountController, CoreId } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+
+export interface DeleteRepositoryAttributeRequest {
+    attributeId: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<DeleteRepositoryAttributeRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("DeleteRepositoryAttributeRequest"));
+    }
+}
+
+export class DeleteRepositoryAttributeUseCase extends UseCase<DeleteRepositoryAttributeRequest, void> {
+    public constructor(
+        @Inject private readonly attributeController: AttributesController,
+        @Inject private readonly accountController: AccountController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: DeleteRepositoryAttributeRequest): Promise<Result<void>> {
+        const repositoryAttributeId = CoreId.from(request.attributeId);
+        const repositoryAttribute = await this.attributeController.getLocalAttribute(repositoryAttributeId);
+
+        if (typeof repositoryAttribute === "undefined") {
+            return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
+        }
+
+        if (!repositoryAttribute.isRepositoryAttribute(this.accountController.identity.address)) {
+            return Result.fail(RuntimeErrors.attributes.isNotRepositoryAttribute(repositoryAttributeId));
+        }
+
+        const ownSharedIdentityAttributeCopies = await this.attributeController.getLocalAttributes({ "shareInfo.sourceAttribute": repositoryAttributeId.toString() });
+        const ownSharedIdentityAttributePredecessors = await this.attributeController.getSharedPredecessorsOfRepositoryAttribute(repositoryAttribute);
+        for (const ownSharedAttribute of [...ownSharedIdentityAttributeCopies, ...ownSharedIdentityAttributePredecessors]) {
+            if (!ownSharedAttribute.isOwnSharedAttribute(this.accountController.identity.address)) {
+                return Result.fail(RuntimeErrors.attributes.isNotOwnSharedAttribute(ownSharedAttribute.id));
```

Comment:
If everything goes right, this error should never be called. We need it, however, so that the compiler knows in the following line that `ownSharedAttribute.shareInfo` is defined and we can set its `sourceAttribute` property.

Reply:
I don't think we should return a result in this case and just throw an error. It's absolutely not expected that this happens (when nobody messes up the data from outside).